nixos/modules/installer/tools/tools.nix

```diff
@@ -157,6 +157,12 @@ in
         #   extraGroups = [ "wheel" ]; # Enable ‘sudo’ for the user.
         # };
 
+        # With this option enabled, if you accidentally remove this file you can
+        # still recover it from
+        # - /run/current-system/configuration.nix (last one)
+        # - /nix/var/nix/profiles/system-*/configuration.nix (old versions)
+        system.copySystemConfiguration = true;
```

Comment:
Should this be false by default?

Reply:
The option is ``false`` by default, but I think the point of this pull request is have it enabled for NixOS.

Reply:
I'm saying this, because originally I've suggested "false" as default.

https://github.com/NixOS/nixpkgs/pull/24707#discussion_r316321687

My reasoning is like this:
- default config should be close to real NixOS defaults.
- ideally, we should take an idea from [postgresql default config](https://github.com/postgres/postgres/blob/REL_11_STABLE/src/backend/utils/misc/postgresql.conf.sample) - list an option, but leave it commented, together with default value. If you uncomment - it is still default, you have both to _uncomment_ and _change_ to make thing happen.
- configuration can have secrets embedded, in which case it shouldn't be copied to store by default. Or at least warned about that. We care about security.

Reply:
> * configuration can have secrets embedded, in which case it shouldn't be copied to store by default. Or at least warned about that. We care about security.

If the secret is embedded and used, it ends up in the nix store anyway, no?

Reply:
@mmahut  :thinking: indeed. Okay, then I'd be fine with commented-by-default, as are other options.

Reply:
I'd like for this to be `true` and uncommented because there's no harm in doing so, the worst thing it can do is increase the closure size a bit, but the benefit it provides will be appreciated by a lot. If problems like the one with nixops wouldn't exist I'd make this option's `default = true;`; changing the default here is the next best thing we can do.

Reply:
I won't take a hard stance and counter this. I just want to keep consistency in the sample config file. The only uncommented options are vital ones - hardware config, boot loader config and state version. Everything else is "recommendation" and thus, commented out.

From pragmatic viewpoint, yes, it would be great to make it enabled for new users.